### DISCUSSION TITLE
Allow branding for devices within hub integrations to improve user experience

### DISCUSSION THREAD

**Discussion by jbouwh** [home-assistant/architecture#1006] [Entity Models] [answered]
## Background

Hub integrations like Z-Wave, ZHA, Matter, Philips Hue or MQTT can connect devices or hubs of different brands.
Currently the discovered devices inherit the branding of the integration. E.g. you can link a IKEA Tradfi bulb to a Philips Hue Bridge, but it still will have the Hue branding.

## Proposal

Allow hub integrations to link devices to a specific within Home Assistant registered brand. The brand reference (a string) would be stored in the Device Registry in case branding is specified. The frontend will update the icon of entities and device image to the linked branding. Devices should still be linked to the integration that discovered and generated them.

## Rationale

Branding will make it easier for users to find their devices because they a recognized.

Manufacturers do not need to come with own integrations if there is already a hub application that already brings the needed functionality.

Example is the PR for the [Drop connect](https://github.com/home-assistant/core/pull/104319) integration, which intents to create branded MQTT discovery.

**Comment by MartinHjelmare** [comment]
Side note: We shouldn't encourage developers to use the MQTT discovery as an API for integration instead of building an integration. Building an integration allows for much better integration than just using MQTT discovery. We should try to limit the pressure to add brand specific features to MQTT discovery keeping maintenance controlled.

MQTT is first and foremost a transport and we should not aim at making it a preferred application protocol in line with Z-Wave, Zigbee and Matter. We shouldn't take on that effort and burden.

For the application protocols Z-Wave, Zigbee and Matter, it sounds nice to be able to show the brand of a device.

**Reply by jbouwh** [reply]
About the side note for MQTT. If we decide to have the ability to show a different brand, for a device then MQTT users / custom services outside HA could also benefit here IMO, for example Zigbee2MQTT. I do not think this will a negative impact on MQTT maintenance as this can be done centralized.

**Reply by MartinHjelmare** [reply]
If we allow MQTT to set a brand for a device we'd need to extend all the config and discovery entry points we have today for MQTT. That's exactly the maintenance I'm talking about and what I don't think we should do.

**Reply by raman325** [reply]
See https://github.com/home-assistant/architecture/discussions/1006#discussioncomment-7789796. I think we can do this in a way that doesn't require any changes to existing models

**Comment by raman325** [comment]
I think it would be a nice user experience improvement to be able to display the brand image on the device info page. I had a couple of thoughts about this proposal:
1. If no brand exists, it should fall back to what we have now, e.g. for a Z-Wave device show the Z-Wave logo.
2. We may want to consider showing both logos (brand + tech/hub) in some way since brands will often support multiple technologies and I do think knowing the technology underlying the integration can sometimes be important for user actions/decisions.
3. We already have the `manufacturer` and `model` field in the `DeviceEntry` model. We should consider how we can reuse that for this solution, which may require some normalization and/or aliasing as brands might use different capitalization, have additional words in the manufacturer name for some devices, etc.
4. Whatever we do, speaking from a biased position of course, I don't think integrations should have to do anything to add support for this. They're already doing the work of collecting the manufacturer name when they can, so, instead of distributing the work of doing the mapping, which would require a lot of duplicate logic, we should centralize it in the core.
5. I am not sure maintaining the mappings in the core code is going to be the best approach. We already have an alternate mechanism for logos (the `brands` repo), perhaps we can use that somehow?
6. This may already be obvious, but it should be a requirement to add logos for any brand we want to map to, otherwise this feature doesn't provide a ton of value.

**Comment by frenck** [comment, marked as answer]
Honestly, I don't think this fits in our current model. The branding used right now, is mostly referring to integration branding. We do not have a notion of actual devices yet (besides it being a grouping).

However, we do have ideas and proposals for adding actual devices at a later stage to Home Assistant (probably somewhere in 2024). Therefore, I think we should not bother with this proposal at this point in time, as it will bring us with model complexity to migrate if we go down the devices path.

Additionally, I agree with Martin, MQTT isn't a good example in this set case. Also, I think that the example from DROP is actually a good example, and not the anti-example as pointed out.


../Frenck
